src/components/Map.tsx

```diff
@@ -0,0 +1,145 @@
+      this.setState({ directionsAreToggled: true });
+
+      directionsDisplay.setMap(map);
+
+      let calculateAndDisplayRoute = (
+        directionsService: any,
+        directionsDisplay: any
+      ) => {
+        directionsService.route(
+          {
+            origin: userLocation,
+            destination: {
+              lat: resource.lat,
+              lng: resource.lng
+            },
+            travelMode: 'TRANSIT'
+          },
+          function(response: any, status: any) {
+            if (status === 'OK') {
+              directionsDisplay.setDirections(response);
+            } else {
+              window.alert('Directions request failed due to ' + status);
+            }
+          }
+        );
+      };
+
+      calculateAndDisplayRoute(directionsService, directionsDisplay);
+    } else {
+      directionsDisplay.setMap(null);
+      this.setState({ directionsAreToggled: false });
+    }
+  };
+
+  createMarkers = (resources: any, map: any, maps: any) => {
+    resources.forEach((resource: any) => {
+      let marker = new maps.Marker({
+        map: map,
+        title: resource.name,
+        position: {
+          lat: resource.lat,
+          lng: resource.lng
+        }
+      });
+
+      let infoWindow = new maps.InfoWindow({
+        content:
+          '<div><b>' +
+          resource.name +
+          '</b><br>' +
+          resource.category +
+          '<br>' +
+          resource.address +
+          '<br><br>' +
+          '<b>For</b>: ' +
+          resource.audience +
+          '<br>' +
+          '<b>Offers</b>: ' +
+          resource.services.join(', ') +
+          '</div>'
+          // will need to include if its open/hours (unsure of how resource.hours will be formatted)
+      });
+
+      marker.addListener('mouseover', function() {
+        infoWindow.open(map, marker);
+      });
+
+      marker.addListener('mouseout', function() {
+        infoWindow.close();
+      });
+    });
+  };
+
+  initMap = (map: any, maps: any) => {
+    let directionsService = new maps.DirectionsService();
+    let directionsDisplay = new maps.DirectionsRenderer();
+    this.setState({
+      map: map,
+      maps: maps,
+      directionsService: directionsService,
+      directionsDisplay: directionsDisplay
+    });
+
+    this.createMarkers(this.props.resources, map, maps);
+  };
+
+  render() {
+    const firstResource = this.props.resources[0];
+    const toggleDirectionsOnClick = () => {
+      this.toggleDirections(firstResource);
+    };
+    let center = {
+      lat: firstResource.lat,
+      lng: firstResource.lng
+    };
+
+    return (
+      // Map must have height/width defined - but manipulate as necessary
+      <div style={{ height: '60vh', width: '100%' }}>
+        <h1>Map</h1>
+        <button onClick={toggleDirectionsOnClick}>
+          Show/Hide Directions to {firstResource.name}
+        </button>
+        <GoogleMapReact
+          bootstrapURLKeys={{ key: '<API_KEY>' }}
```

Comment:
While I haven't tried it myself, I think we can create a local environment variable for this to make updating API keys easier. You could replace `'<API_KEY>'` with `{REACT_APP_API_KEY_GOOGLE_MAPS}` and then add that variable to your own [process.env](https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables#what-other-env-files-can-be-used) that sits at the top level of this project.

I'm thinking for local dev, it would be best to use our own API keys and then have our centralized one on the live site.

Reply:
Sounds like a good approach.  I'm struggling to set that up - I've added it to my .env and I've tried running `require('dotenv').config()` before calling on it, but I think I'm missing something.

Reply:
I'll pull the branch down in a bit to see if I can get it working with my own API key.